### ISSUE TITLE
[documentation] clarification about the signal safe stack tracing

### ISSUE THREAD

**Issue by bertulli** [jeremy-rifkin/cpptrace#146]
Hi, first of all thanks for your work. I was trying to integrate cpptrace in my project to investigate signal faults (specifically SIGSEGV), and I noticed a few things:
- the document [on signal safe stack tracing](https://github.com/jeremy-rifkin/cpptrace/blob/main/docs/signal-safe-tracing.md) mentions two files (`signal_demo.cpp` and `signal_tracer.cpp`), that are broken links
- in the example code, there is this for loop:
```c
for(std::size_t i = 0; i < count; i++) {
        cpptrace::safe_object_frame frame;
        cpptrace::get_safe_object_frame(buffer[i], &frame);
        write(input_pipe.write_end, &frame, sizeof(frame));
    }
```
but the variable `count` is not defined. Since it is the limit of `buffer`, a `frame_ptr`, how should I extract the size of it?

As a side note, it's mentioned in #126 that older versions of glibc can't print the traces: do you know what is the minimum version?

Thanks!

**Comment by jeremy-rifkin** [comment]
Hi, thanks for pointing out the broken links. I’ll fix them tonight (or accept a PR), the files are in https://github.com/jeremy-rifkin/cpptrace/tree/main/test. The count in question comes from the generate safe raw trace function,
```cpp
std::size_t count = cpptrace::safe_generate_raw_trace(…);
```

The minimum glibc version is 2.35, released in early 2022. Unfortunately it’s a relatively high glibc requirement but _dl_find_object is the only way to do this in a truly signal-safe way, as far as I have found.

**Comment by jeremy-rifkin** [comment]
I've pushed some fixes to the documentation in e8fce3fad1d299bb4effbe5af701bbc9bbd10ab5
